x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/text/TextTemplate.java

```diff
@@ -48,12 +50,15 @@ public TextTemplate(String template, @Nullable XContentType contentType, ScriptT
             params = new HashMap<>();
         }
         this.script = new Script(type, type == ScriptType.STORED ? null : Script.DEFAULT_TEMPLATE_LANG, template, options, params);
+        this.isUsingMustache = template.contains("{{") && template.indexOf("{{") < template.indexOf("}}");
         this.inlineTemplate = null;
     }
 
     public TextTemplate(Script script) {
         this.script = script;
         this.inlineTemplate = null;
+        this.isUsingMustache = script.getIdOrCode().contains("{{") &&
```

Comment:
can we lazy load this in the getter? maybe remove the word `get` so we know it "mutates state" and load it once only if we decide to actually validate if we care or not. At least it will appease the `contains`'ing every string is bad gods.

Reply:
I changed this only to check for `{{`, and it #33978 will introduce a check for every text template, so that this will be called anyway...

Reply:
oof, how did i miss that? I guess going backwards thru my review-requested emails is not such a good idea!!